presto-main/src/main/java/com/facebook/presto/execution/executor/TaskHandle.java

```diff
@@ -160,6 +208,34 @@ public synchronized PrioritizedSplitRunner pollNextSplit()
         return split;
     }
 
+    private boolean isAnySplitStarted(Queue<PrioritizedSplitRunner> queuedLeafSplits)
+    {
+        for (PrioritizedSplitRunner splitRunner : queuedLeafSplits) {
+            if (splitRunner.isSplitAlreadyStarted()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void gracefulShutdown()
+    {
+        isShuttingDown.set(true);
+    }
```

Comment:
nit calling it graceful shutdown implies there is some action to be done.
maybe just call it as markTaskAsShutdown